katz/src/main/kotlin/katz/data/Option.kt

```diff
@@ -31,6 +31,7 @@ sealed class Option<out A> : OptionKind<A> {
     companion object : OptionMonad, GlobalInstance<Monad<Option.F>>() {
         inline fun <reified A : Any> fromNullable(a: A?): Option<A> = if (a != null) Option.Some(a) else Option.None
         operator fun <A> invoke(a: A): Option<A> = Option.Some(a)
+        inline fun <reified T: Any> empty() = Option.fromNullable<T>(null)
```

Comment:
This is a way to create and bind a type to the Option without having the initial value ready yet. This method is also available in Scala Options and helps you on type inference for some given cases